workflow/rules/scan_decomp_data.smk

```diff
@@ -0,0 +1,9 @@
+rule scan_decomp_dir:
+    input:
+        directory("results/uncompressed-docs/{id}")
```

Comment:
I recall, that the directory() flag is primarily designed for Snakemake outputs: https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#directories-as-outputs

However, since it works as input, this is just a sidenote.